### PR TITLE
[MOD-3974] Fix grouped functions

### PR DESCRIPTION
**Pull request by thecodingwizard**
This PR fixes grouped functions introduced in #2212 and adds a test.

Previously, I couldn't get grouped functions to work. A simple function like this:

```py
app = modal.App("hello-world")
@app.function()
@modal.experimental.grouped(size=2)
def simple():
    print("simple")

@app.local_entrypoint()
def main():
    simple.remote()
```

would fail with

```
Traceback (most recent call last):
  File "/pkg/modal/_container_io_manager.py", line 735, in handle_user_exception
    yield
  File "/pkg/modal/_container_entrypoint.py", line 843, in main
    finalized_functions = service.get_finalized_functions(function_def, container_io_manager)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pkg/modal/_container_entrypoint.py", line 137, in get_finalized_functions
    is_async = get_is_async(self._user_defined_callable)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pkg/modal/_utils/function_utils.py", line 93, in is_async
    raise RuntimeError(f"Function {function} is a strange type {type(function)}")
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: __str__ returned non-string (type function)
```

I think `function` is supposed to be a normal function but is instead a `_GroupedFunction`.

### CODE REVIEW

File: test/conftest.py
Comment (thecodingwizard) [inline test/conftest.py:244]: This change is needed because grouped functions would pass in a `modal_q` kwarg that is a modal Queue rather than a number, which would cause issues.

File: test/conftest.py
Comment (thundergolfer) [inline test/conftest.py:242]: instead of making this default more complex could you instead set `function_body` below just for the grouped function?

File: test/function_test.py
Comment (thundergolfer) [inline test/function_test.py:1004]: Thanks for adding the test. definitely need to stop regressions happened while we iterate. 

There's already `i6pn_group_test.py` which is a good place to slot this in.

**Review by thundergolfer — APPROVED** [review]
LGTM if tests are passing. left couple suggestions

**Comment by thecodingwizard** [comment]
I think I fixed the unit tests by monkeypatching modal.Queue.ephemeral to accept the right client, but is there a neater way to do this?

**Comment by thecodingwizard** [comment]
CI is blocked by https://github.com/modal-labs/synchronicity/pull/172

**Pull request merged**
